pkg/controllers/sync/resource.go

```diff
@@ -56,94 +51,78 @@ type FederatedResource interface {
 	dispatch.FederatedResourceForDispatch
 
 	FederatedName() common.QualifiedName
-	FederatedKind() string
-	FederatedGVK() schema.GroupVersionKind
-	CollisionCount() *int32
-	RevisionHistoryLimit() int64
 	UpdateVersions(selectedClusters []string, versionMap map[string]string) error
 	DeleteVersions()
-	ComputePlacement(clusters []*fedcorev1a1.FederatedCluster) (selectedClusters sets.String, err error)
-	NamespaceNotFederated() bool
+	ComputePlacement(clusters []*fedcorev1a1.FederatedCluster) sets.Set[string]
+	SetObject(obj fedcorev1a1.GenericFederatedObject)
 }
 
+var _ FederatedResource = &federatedResource{}
+
```

Comment:
```suggestion
var _ version.VersionedResource = &federatedResource{}
```

Reply:
Should `FederatedResource` also implement `version.VersionedResource` ?

Reply:
Yes I think so